packages/remix-dev/codemod/replace-remix-magic-imports/transform.ts

```diff
@@ -0,0 +1,229 @@
+  program.traverse({
+    ImportDeclaration: (path) => {
+      let source = path.node.source.value;
+      if (!isRemixPackage(source)) {
+        return;
+      }
+      found.push(path);
+    },
+  });
+  return found;
+};
+
+const flattenRemixImportDeclarations = (
+  remixImportDeclarations: NodePath<t.ImportDeclaration>[]
+): ExistingExport<RemixPackage>[] => {
+  let flattened: ExistingExport<RemixPackage>[] = [];
+  remixImportDeclarations.forEach((declaration) => {
+    let { source, specifiers } = declaration.node;
+    // Side-effect imports like `import "remix"` not produced flattened specifiers
+    // so they will not contribute to the new Remix import declarations
+    specifiers.forEach((specifier) => {
+      if (t.isImportDefaultSpecifier(specifier)) {
+        throw declaration.buildCodeFrameError(
+          "This codemod does not support default imports for the `remix` package.\n" +
+            "Replace the default import with named imports and try again."
+        );
+      }
+      if (t.isImportNamespaceSpecifier(specifier)) {
+        throw declaration.buildCodeFrameError(
+          "This codemod does not support namespace imports for the `remix` package.\n" +
+            "Replace the namespace import with named imports and try again."
+        );
+      }
+      let { imported, local, importKind } = specifier;
+
+      let kind =
+        importKind === "type" || declaration.node.importKind === "type"
+          ? ("type" as const)
+          : importKind ?? declaration.node.importKind ?? "value";
+
+      let name = t.isStringLiteral(imported) ? imported.value : imported.name;
+
+      flattened.push({
+        source: source.value as RemixPackage,
+        kind,
+        name,
+        alias: local.name === name ? undefined : local.name,
+        _declaration: declaration,
+      });
+    });
+  });
+  return flattened;
+};
+
+const convertToNewRemixImports = (
+  currentRemixImports: ExistingExport<RemixPackage>[],
+  options: Options
+): Export<Exclude<RemixPackage, "remix">>[] => {
+  let replaceRemixMagicImport = createRemixMagicImportReplacer(options);
+  let newRemixImports = currentRemixImports.map((currentRemixImport) => {
+    if (currentRemixImport.source !== "remix") {
+      return currentRemixImport as Export<Exclude<RemixPackage, "remix">>;
+    }
+
+    let newRemixImport = replaceRemixMagicImport({
+      kind: currentRemixImport.kind,
+      name: currentRemixImport.name,
+    });
+    if (newRemixImport === undefined) {
+      throw currentRemixImport._declaration.buildCodeFrameError(
+        `Unrecognized import from 'remix': ` +
+          `${currentRemixImport.kind === "type" ? "type " : ""}${
+            currentRemixImport.name
+          }`
+      );
+    }
+    return {
+      source: newRemixImport.source,
+      kind: currentRemixImport.kind,
+      name: newRemixImport.name,
+      alias: currentRemixImport.alias,
+    };
+  });
+  return newRemixImports;
```

Comment:
Since we're only returning `newRemixImports`, we can do that directly instead of having an unnecessary local variable

```suggestion
```